src/main/clojure/clara/tools/inspect.cljc

```diff
@@ -4,16 +4,24 @@
    * inspect, which returns a data structure describing the session that can be used by tooling.
    * explain-activations, which uses inspect and prints a human-readable description covering
      why each rule activation or query match occurred."
-  (:require [clara.rules.engine :as eng]
+  (:require #?(:clj [clara.rules.engine :as eng])
+            #?(:cljs [clara.rules.engine :as eng :refer [RootJoinNode
+                                                         HashJoinNode
+                                                         ExpressionJoinNode
+                                                         NegationNode
+                                                         NegationWithJoinFilterNode]])
             [clara.rules.schema :as schema]
             [clara.rules.memory :as mem]
             [schema.core :as s])
-  (:import [clara.rules.engine
-            RootJoinNode
-            HashJoinNode
-            ExpressionJoinNode
-            NegationNode
-            NegationWithJoinFilterNode]))
+  #?(:clj
+    (:import [clara.rules.engine
+              RootJoinNode
+              HashJoinNode
+              ExpressionJoinNode
+              NegationNode
+              NegationWithJoinFilterNode]))
+  #?(:cljs
+     (:import goog.string)))
```

Comment:
Although this tends to work, the proper cljs way to load in a closure library is via `:require`.
Along with that, I'd recommend a ns alias to refer to symbols from it.
```clj
(:require [goog.string :as gstr])

;; later

(gstr/startsWith (name k) "?__gen__")
```

See http://clojurescriptmadeeasy.com/blog/when-do-i-use-require-vs-import.html

Reply:
@sparkofreason 

The issue I logged at #363 is related to this, but on the CLJ-side of things instead:

Clara is intentionally building and supporting an older version of CLJ in order to just have broader reach. It is unfortunate though when it makes certain features not exists yet, like `clojure.string/starts-with?` (along with probably quite a bit of fn's designed to make writing CLJC easier/less-host-dependent/with less read-conditionals.
However, I do think it'd be good to have profiles for newer versions of CLJ to ensure Clara is building and passing tests against them.